CorsixTH/Lua/map.lua

```diff
@@ -252,7 +252,14 @@ function Map:load(level, difficulty, level_name, map_file, level_intro, map_edit
       return nil, errors
     end
     assert(base_config, "No base config has been loaded!")
-    errors, result = self:loadMapConfig(self.app:getAbsolutePathToLevelFile(level), base_config, true)
+    local search_paths = {
+      self.app.user_campaign_dir,
+      self.app.user_level_dir,
+      self.app.campaign_dir,
+      self.app.level_dir,
+    }
+    local filename = self.app:findFileInDirs(search_paths, level)
+    errors, result = self:loadMapConfig(filename, base_config, true)
```

Comment:
You can load a config from a different spot than the map file. That makes it difficult to predict what files are going to be used.

In my view, if you load a campaign from `self.app.campaign_dir` the map and the config should also come from there.

Not sure what should happen if some files are missing, but things should be predictable for a user and level designer.

Also, preferably it should be stable under adding other files (but not sure this is achievable). Eg if a level file was released and it didn't have config, we used file X. If I then add a new version of the level and that one does have a config, then the new version should use that config, but the old one perhaps not, since the entire experience of the level could change.